ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/XATerminatorImple.java

```diff
@@ -452,7 +453,12 @@ public boolean beforeCompletion (Xid xid) throws javax.transaction.SystemExcepti
 
     public Transaction getTransaction(Xid xid) throws XAException {
         // first see if the xid is a root coordinator
-        return com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.getTransaction(new XidImple(xid).getTransactionUid());
+        Transaction transaction = TransactionImple.getTransaction(new XidImple(xid).getTransactionUid());
```

Comment:
As mentioned on spi comment it's not possible as WFTC uses this to integrate with Narayana. I tested that with JTS and JTA on the failing cases and it worked.
If the functionality should be removed we need to think about different way for WFTC to integrate.